proto/destination.proto

```diff
@@ -166,6 +172,47 @@ message ProtocolHint {
   }
 }
 
+// Configures the parameters used to initialize an HTTP/2 connection.
+message Http2ClientParams {
+  // Overrides the default client flow control settings.
+  FlowControl flow_control = 1;
+
+  // Enables keep-alive timeouts.
+  KeepAlive keep_alive = 2;
+
+  // Configures Hyper internals.
+  Internals internals = 3;
+
+  message FlowControl {
+    // Configures the maximum connection-level flow control window size.
+    uint32 initial_connection_window_size = 1;
+
+    // Configures the maximum stream-level flow control window size.
+    uint32 initial_stream_window_size = 2;
+
+    // Enables Hyper's adaptive flow control, ignoring other window settings.
+    bool adaptive_flow_control = 3;
+  }
+
+  message KeepAlive {
+    // The time between pings.
+    google.protobuf.Duration interval = 1;
+
+    // The time to wait for a ping response before considering the connection
+    // dead.
+    google.protobuf.Duration timeout = 2;
+
+    // Whether to send pings when there is no other traffic.
+    bool while_idle = 3;
```

Comment:
what are the semantics of this? if false will pings be send when there IS other traffic? if true will pings ONLY be sent when there is no other traffic?

Reply:
From [hyper](https://docs.rs/hyper/latest/hyper/client/conn/http2/struct.Builder.html#method.keep_alive_while_idle)
> Sets whether HTTP2 keep-alive should apply while the connection is idle.
>
> If disabled, keep-alive pings are only sent while there are open request/responses streams. If enabled, pings are also sent when no streams are active. Does nothing if keep_alive_interval is disabled.